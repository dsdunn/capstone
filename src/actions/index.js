export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const updateUser = (user) => ({
  type: 'UPDATE_USER',
  user
})

export const setProfile = profile => ({
  type: 'SET_PROFILE',
  profile
})

export const addCollections = (collections) => ({
  type: 'ADD_COLLECTIONS',
  collections
})

export const setCollection = collection => ({
  type: 'SET_COLLECTION',
  collection
})

export const updateCollectionsList = (collectionsList) => {
  type: 'UPDATE_COLLECTIONS_LIST',
  collectionsList
}