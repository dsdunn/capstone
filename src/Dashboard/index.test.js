
import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard, mapDispatchToProps, mapStateToProps, } from './'
import { setProfile } from '../actions';

describe('Dashboard', () => {
  let wrapper;
  let mockSetProfile;
  let mockUser;
  let mockHistory;
  let mockHandleSignOut;

  beforeEach(() => {
    mockHistory = { 
                   location: { pathname: '/'},
                   goBack: jest.fn(),
                   push: jest.fn() 
                  }

    mockSetProfile = jest.fn()
    mockHandleSignOut = jest.fn()

    mockUser = {
                avatar: '2 comes out in 2020',
                username: 'username? username for what?',
                location: 'somewhere else',
                bio: 'dome'
               }

    wrapper = shallow(<Dashboard setProfile={mockSetProfile} 
                                 user={mockUser}
                                 history={mockHistory}
                                 handleSignOut={mockHandleSignOut}
                                  />)
  })

  it('should call setProfile when handleViewProfile is called', () => {
    wrapper.find('.dashboard-view-profile').simulate('click')
    expect(mockSetProfile).toHaveBeenCalled()
  })

  it('should route the user to a new destination when getPath is called', () => {
    wrapper.find('.test3').simulate('click')
    expect(mockHistory.push).toHaveBeenCalled()
  })

  it('should call handleSignOut when Sign out btn is clicked', () => {
    wrapper.find('.test4').simulate('click')
    expect(mockHandleSignOut).toHaveBeenCalled()
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should return a props object with setProfile', () => {
    const mockSignIn = jest.fn();
    const mockDispatch = jest.fn();

    const mappedProps = mapDispatchToProps(mockDispatch)
    const actionToDispatch = setProfile(mockUser);

    mappedProps.setProfile(mockUser);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
  it('should return a props object', () => {
    const mockState = {user: {}};
    const expected = {"user": {}};

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  })
})
