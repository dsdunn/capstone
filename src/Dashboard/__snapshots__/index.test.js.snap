// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addCollections={[Function]}
    handleSignOut={[Function]}
    history={
        Object {
            "goBack": [Function],
            "location": Object {
              "pathname": "/",
            },
            "push": [Function],
          }
    }
    setProfile={[Function]}
    user={
        Object {
            "avatar": "2 comes out in 2020",
            "bio": "dome",
            "location": "somewhere else",
            "username": "username? username for what?",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="dashboard-username"
>
          Hello, 
          username? username for what?
          .
</p>,
        <div>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="dashboard-link dashboard-view-profile"
                    exact={true}
                    onClick={[Function]}
                    to="/user"
          >
                    View Profile
          </NavLink>
          <a
                    className="dashboard-link test3"
                    onClick={[Function]}
          >
                    Edit Profile
          </a>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="dashboard-link"
                    to="/user/addcollection"
          >
                    Add Collection
          </NavLink>
          <Link
                    className="dashboard-link test4"
                    onClick={[Function]}
                    replace={false}
                    to="/"
          >
                    Sign Out
          </Link>
</div>,
      ],
      "className": "dashboardHide dashboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hello, ",
            "username? username for what?",
            ".",
          ],
          "className": "dashboard-username",
        },
        "ref": null,
        "rendered": Array [
          "Hello, ",
          "username? username for what?",
          ".",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="dashboard-link dashboard-view-profile"
              exact={true}
              onClick={[Function]}
              to="/user"
>
              View Profile
</NavLink>,
            <a
              className="dashboard-link test3"
              onClick={[Function]}
>
              Edit Profile
</a>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="dashboard-link"
              to="/user/addcollection"
>
              Add Collection
</NavLink>,
            <Link
              className="dashboard-link test4"
              onClick={[Function]}
              replace={false}
              to="/"
>
              Sign Out
</Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "View Profile",
              "className": "dashboard-link dashboard-view-profile",
              "exact": true,
              "onClick": [Function],
              "to": "/user",
            },
            "ref": null,
            "rendered": "View Profile",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edit Profile",
              "className": "dashboard-link test3",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Edit Profile",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Add Collection",
              "className": "dashboard-link",
              "to": "/user/addcollection",
            },
            "ref": null,
            "rendered": "Add Collection",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Sign Out",
              "className": "dashboard-link test4",
              "onClick": [Function],
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Sign Out",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="dashboard-username"
>
            Hello, 
            username? username for what?
            .
</p>,
          <div>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="dashboard-link dashboard-view-profile"
                        exact={true}
                        onClick={[Function]}
                        to="/user"
            >
                        View Profile
            </NavLink>
            <a
                        className="dashboard-link test3"
                        onClick={[Function]}
            >
                        Edit Profile
            </a>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="dashboard-link"
                        to="/user/addcollection"
            >
                        Add Collection
            </NavLink>
            <Link
                        className="dashboard-link test4"
                        onClick={[Function]}
                        replace={false}
                        to="/"
            >
                        Sign Out
            </Link>
</div>,
        ],
        "className": "dashboardHide dashboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hello, ",
              "username? username for what?",
              ".",
            ],
            "className": "dashboard-username",
          },
          "ref": null,
          "rendered": Array [
            "Hello, ",
            "username? username for what?",
            ".",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="dashboard-link dashboard-view-profile"
                exact={true}
                onClick={[Function]}
                to="/user"
>
                View Profile
</NavLink>,
              <a
                className="dashboard-link test3"
                onClick={[Function]}
>
                Edit Profile
</a>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="dashboard-link"
                to="/user/addcollection"
>
                Add Collection
</NavLink>,
              <Link
                className="dashboard-link test4"
                onClick={[Function]}
                replace={false}
                to="/"
>
                Sign Out
</Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "View Profile",
                "className": "dashboard-link dashboard-view-profile",
                "exact": true,
                "onClick": [Function],
                "to": "/user",
              },
              "ref": null,
              "rendered": "View Profile",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit Profile",
                "className": "dashboard-link test3",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Edit Profile",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Add Collection",
                "className": "dashboard-link",
                "to": "/user/addcollection",
              },
              "ref": null,
              "rendered": "Add Collection",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sign Out",
                "className": "dashboard-link test4",
                "onClick": [Function],
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Sign Out",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
